File: _a_arm/matlab/hw16/armParamHW16.m
% single link arm parameter file
addpath ./.. % adds the parent directory to the path
armParam % general parameters

% load parameters from HW 10
addpath ./../hw10
armParamHW10

% transfer function for robot arm
th_e = 0;
G = tf([2/P.m/P.ell^2],[1, 2*P.b/P.m/P.ell^2, -3*P.g*sin(th_e)/2/P.ell]);
figure(3), clf, bode(G), grid on

C_pid = tf([(P.kd+P.kp*P.sigma),(P.kp+P.ki*P.sigma),P.ki],[P.sigma,1,0])
figure(4), clf, bode(G), grid on
hold on
bode(series(G,C_pid))
legend('plant', 'PID control')

File: _A_arm/python/hw16/armParamHW16.py
# Single link arm Parameter File
import sys
sys.path.append('..')  # add parent directory
import armParam as P
sys.path.append('../hw10')  # add parent directory
sys.path.append('../hw15')  # add parent directory
import armParamHW10 as P10
import armParamHW15 as P15
from control import tf, bode
import matplotlib.pyplot as plt


# flag to define if using dB or absolute scale for M(omega)
dB_flag = P15.dB_flag

# Assign plan from previous homework solution
Plant = P15.Plant

# Compute transfer function of controller
C_pid = tf([(P10.kd+P10.kp*P.sigma), (P10.kp+P10.ki*P.sigma), P10.ki],
           [P.sigma, 1, 0])

if __name__=="__main__":
    # display bode plots of transfer functions
    fig = plt.figure()
    bode([Plant, Plant*C_pid], dB=dB_flag)
    fig.suptitle('Single Link Arm')
    plt.legend(['P(s)', 'C(s)P(s)'])
    print('Close plot window to end program')
    plt.show()
